### PR TITLE
Fix hostname not being set for worker nodes in freestyle jobs

### PR DESCRIPTION
**Pull request by albertvaka**
### What does this PR do?

Try to read hostname from the environment if not in `updatedBuildData`.

If a job has no child stages (eg: in a freestyle job), `updatedBuildData` in `DatadogWebhookBuildLogic` will never actually be updated because `DatadogWebhookPipelineLogic` didn't run. Because of this even if the job runs in a worker node everything we read from `updatedBuildData` will be the main node's info. 

When this happens, this PR adds a workaround to read the hostname from the `DD_CI_HOSTNAME` env var directly from `DatadogWebhookBuildLogic`.

### Alternate Designs

Ideally we should have a single way to get the workers info that works both for pipelines and for single jobs. But the logic is too convoluted and I'm not confident enough to do a bigger change.

### Verification Process

Run a freestyle job on a worker with the `DD_CI_HOSTNAME` variable set, check that the host is set properly in the payloads.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: src/main/java/org/datadog/jenkins/plugins/datadog/model/StepData.java [hunk 1]
```diff
@@ -19,7 +19,7 @@ public class StepData implements Serializable {
     private static final long serialVersionUID = 1L;
 
     // Env Var key to get the hostname from the Jenkins workers.
-    private static final String DD_CI_HOSTNAME = "DD_CI_HOSTNAME";
+    public static final String DD_CI_HOSTNAME = "DD_CI_HOSTNAME";
```
Comment (drodriguezhdez) [inline src/main/java/org/datadog/jenkins/plugins/datadog/model/StepData.java:22]: Maybe worth it to move this to a shared class.

File: src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java [hunk 2]
```diff
@@ -142,10 +146,18 @@ public void finishBuildTrace(final BuildData buildData, final Run<?,?> run) {
...
             final String workerHostname = getNodeHostname(run, updatedBuildData);
-            // If the worker hostname is equals to controller hostname but the node name is not "master"
-            // then we could not detect the worker hostname properly. We set _dd.hostname to 'none' explicitly.
+            // If the worker hostname is equals to controller hostname but the node name is not master/built-in then we
+            // could not detect the worker hostname properly. Check if it's set in the environment, otherwise set to none.
             if(buildData.getHostname("").equalsIgnoreCase(workerHostname)) {
-                buildSpan.putMeta(CITags._DD_HOSTNAME, HOSTNAME_NONE);
+                String envHostnameOrNone = HOSTNAME_NONE;
+                try {
+                    Map<String, String> env = run.getEnvironment(new LogTaskListener(logger, Level.INFO));
```
Comment (drodriguezhdez) [inline src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java:154]: This is trick to have the env from the worker? I didn't have this `LogTaskListener` class in my radar.

File: src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java [hunk 3]
```diff
@@ -142,10 +146,18 @@ public void finishBuildTrace(final BuildData buildData, final Run<?,?> run) {
...
+                    String envHostname = env.get(StepData.DD_CI_HOSTNAME);
+                    if (StringUtils.isNotEmpty(envHostname)) {
+                        envHostnameOrNone = envHostname;
+                    }
+                } catch (IOException | InterruptedException e) { }
+                buildSpan.putMeta(CITags._DD_HOSTNAME, envHostnameOrNone);
```
Comment (drodriguezhdez) [inline src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java:160]: Could we use `Optional` instead? 

```
var opt = Optional.ofNull("xxxx")
opt.orElse(HOSTNAME_NONE)
```

Comment (albertvaka) [reply src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java:149]: Done.

File: src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java [hunk 4]
```diff
@@ -21,11 +23,13 @@
 import org.datadog.jenkins.plugins.datadog.model.CIGlobalTagsAction;
 import org.datadog.jenkins.plugins.datadog.model.PipelineQueueInfoAction;
 import org.datadog.jenkins.plugins.datadog.model.StageBreakdownAction;
+import org.datadog.jenkins.plugins.datadog.model.StepData;
```
[the remark this answers is not in the document]
Comment (albertvaka) [reply src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java]: Done.

**Review by drodriguezhdez — APPROVED** [review]

**Pull request merged**